src/main/java/com/car/app/boundary/CarResource.java

```diff
@@ -37,9 +39,14 @@
 	private ICarService carService;
 	
 	@GET
-    public Response getCars() {
-        List<CarDto> cars = carService.getCars();
-		return Response.status(Status.OK).entity(cars).build();
+    public Response getCars(@QueryParam("size") int size,
+    		@QueryParam("page") int page,
+    		@QueryParam("sort") String sort,
+    		@QueryParam("order") String order,
```

Comment:
Please perform some validations in the fields you're using for sorting and paging.
You've created a very complete interface for sorting and paging, but I would suggest to simply it a bit, maybe keep only page, order and search to search on a fixed field, for example brand.